contents/java/JUnit5.md

```diff
@@ -0,0 +1,137 @@
+With IntelliJ, you can:
+* Run tests by simply clicking a button
+* Run a subset of tests, to test specific functionality
+* Analyse <popover effect="fade" content="percentage of code checked by tests" placement="top">code coverage</popover>
+* Compare the difference between expected and actual result
+
+### Benefit: JUnit 5 is an [open source project](https://github.com/junit-team/junit5/)
+Being an open source project, JUnit 5 actively accepts contributions from developers. New features are released on cycles averaging every 3-5 weeks. The fast development cycle means that JUnit can constantly keep up with new Java features.
+
+
+## Using JUnit 5: An introduction
+The centerpiece of JUnit 5 is its testing framework. As a test runner, JUnit 5 helps to run your tests.
+
+Let's first start with a sample <popover effect="fade" content="system under test" placement="top">SUT</popover>:
+
+```java
+public class Repeater {
+  // Returns copies of reference string separated by the '\n' character.
+  public static String repeat(String reference, int copies) {
+    String result = "";
+    for (int i = 0; i <= copies; i++) {
+      result += reference + "\n";
+    }
+    return result;
+  }
+}
+```
+
+Let's write a test to check the corectness of our code. By convention, we append `Test` to the name of the tested class.
+
+```java
+public class RepeaterTest {
+  @Test
+  public void repeatTest() {
+    assertEquals(repeat("a", 5), "a\na\na\na\na\n");
+  }
+}
+```
+
+That's it! You do not have to write any additional runners or helpers to take advantage of the powerful testing features. The `@Test` tag tells JUnit 5 that the repeatTest method is meant to be a test. JUnit 5 will then automatically execute your test(s) in sequence.
+
+Let's have a look at the test results. Hmm, the test seems to fail when we run it.
+But did you notice the subtle bug in Repeater?
+
+We "accidentally" looped for an extra time! Without the use of unit tests, we would have to manually confirm that the output indeed has 5 lines, an action that is tedious to repeat and prone to human errors.
+
+Now, let's fix the code up.
+
+```java
+public class Repeater {
+  // Returns copies of reference string separated by the '\n' character.
+  public static String repeat(String reference, int copies) {
+    String result = "";
+    for (int i = 0; i < copies; i++) {
+      result += reference + "\n";
+    }
+    return result;
+  }
+}
+```
+
+Running the same unit test again, we now have a passing test and we can be <trigger for="modal:somewhatAssured" trigger="click">somewhat assured</trigger> that our repeat function works as intended.
+
+## JUnit APIs
+What else can we test with JUnit? In addition to the `assertEquals` method illustrated above, JUnit allows us to test for many other scenarios. These include
+* assertTrue/assertFalse: generic checking statements for checking validity.
```

Comment:
Using inline code highlighting for `assertTrue`, `assertFalse` etc. would be better as they are `JUnit 5` methods.